internal/proxy/impl.go

```diff
@@ -2450,6 +2450,72 @@ func (node *Proxy) Search(ctx context.Context, request *milvuspb.SearchRequest)
 	ctx, sp := otel.Tracer(typeutil.ProxyRole).Start(ctx, "Proxy-Search")
 	defer sp.End()
 
+	if request.SearchByPrimaryKeys {
+		placeholderGroup := &commonpb.PlaceholderGroup{}
+		err := proto.Unmarshal(request.PlaceholderGroup, placeholderGroup)
+		if err != nil {
+			return nil, err
+		}
+
+		queryExpr := string(placeholderGroup.Placeholders[0].Values[0])
+
+		annsField, err := funcutil.GetAttrByKeyFromRepeatedKV(AnnsFieldKey, request.SearchParams)
+		if err != nil {
+			return nil, err
+		}
+
+		queryRequest := &milvuspb.QueryRequest{
+			Base:                  request.Base,
+			DbName:                request.DbName,
+			CollectionName:        request.CollectionName,
+			Expr:                  queryExpr,
+			OutputFields:          []string{annsField},
+			PartitionNames:        request.PartitionNames,
+			TravelTimestamp:       request.TravelTimestamp,
+			GuaranteeTimestamp:    request.GuaranteeTimestamp,
+			QueryParams:           nil,
+			NotReturnAllMeta:      request.NotReturnAllMeta,
+			ConsistencyLevel:      request.ConsistencyLevel,
+			UseDefaultConsistency: request.UseDefaultConsistency,
+		}
+
+		queryResults, err := node.Query(ctx, queryRequest)
+
+		queryResponseStatus := queryResults.Status
+		if queryResponseStatus == nil || queryResponseStatus.ErrorCode != commonpb.ErrorCode_Success {
+			return &milvuspb.SearchResults{
+				Status: &commonpb.Status{
+					ErrorCode: commonpb.ErrorCode_UnexpectedError,
+					Reason:    "query failed in search by primary keys",
+				},
+			}, nil
+		}
```

Comment:
this could be simplified as:
```
err := merr.Error(queryResults.GetStatus())
if err != nil {
    return &milvuspb.SearchResults{
			Status: merr.Status(err),
		}, nil
}
```